R/extremes-data.R

```diff
@@ -135,6 +135,31 @@ createDataRows <- function(data, param, rowName, isUpchain, includeRelated=TRUE,
         dataRows <- dataRows[!duplicated(dataRows[c("date", "primary")]),]
       }
 
+      #Keep only the row with the min/max corresponding value for a given day
+      if(includeRelated && doMerge){
+        if(!isUpchain){
+          #Keep Maximum or Minimum based on current param
+          if(param == "max"){
+            duplicateRows <- dataRows[with(dataRows, order(dataRows$date, dataRows$related, decreasing = TRUE)),]
```

Comment:
Using the function `with` means you don't need to explicitly say that the `date` column is from dataRows. So you can delete the `with` part or delete the `dataRows$` part. It's just redundant right now.

`duplicateRows <- dataRows[with(dataRows, order(date, related, decreasing = TRUE)),]`

or 

`duplicateRows <- dataRows[order(dataRows$date, dataRows$related, decreasing = TRUE),]`